### PR TITLE
Fix AwsInstanceMetadata decoding

### PR DESCRIPTION
**Pull request by msosnicki**
## PR Checklist (not all items are relevant to all PRs)

- [x] Added unit-tests (for runtime code)
- [ ] Added bootstrapped code + smoke tests (when the rendering logic is modified)
- [ ] Added build-plugins integration tests (when reflection loading is required at codegen-time)
- [ ] Added alloy compliance tests (when simpleRestJson protocol behaviour is expanded/updated)
- [ ] Updated dynamic module to match generated-code behaviour
- [ ] Added documentation
- [ ] Updated changelog

### CODE REVIEW

File: modules/aws-http4s/test/src/smithy4s/aws/AwsInstanceMetadataTest.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    internals.AwsJsonCodecs.jsonDecoders.fromSchema(AwsInstanceMetadata.schema)
+
+  test("decodes AwsInstanceMetadata correctly") {
+    val result =
+      codec.decode(Blob("""|{
+                           |  "RoleArn":"arn:aws:iam::123:user:JohnDoe",
+                           |  "AccessKeyId":"access-key",
+                           |  "SecretAccessKey":"secret-key",
+                           |  "Token":"token",
+                           |  "Expiration":"2023-03-03T17:56:46Z"
```
Comment (daddykotex) [inline modules/aws-http4s/test/src/smithy4s/aws/AwsInstanceMetadataTest.scala:18]: To be clear, the issue was that the schema did not define the timestamp format and the AwsInstanceMetadata uses `date-time`

correct?

Comment (msosnicki) [reply modules/aws-http4s/test/src/smithy4s/aws/AwsInstanceMetadataTest.scala:18]: Yes, it's not well documented (at least I cannot find anything), but that's the behavior I'm observing.

Please see https://repost.aws/questions/QUgcf1EIOPS7GZNboeAiyO9Q/renewing-aws-credentials

Comment (Baccata) [reply modules/aws-http4s/test/src/smithy4s/aws/AwsInstanceMetadataTest.scala:18]: @msosnicki that's good. Could you inline a comment in the code with that link, and put a line with the changelog for the next release ?

Comment (msosnicki) [reply modules/aws-http4s/test/src/smithy4s/aws/AwsInstanceMetadataTest.scala:18]: Sure, done

**Review by kubukoz — APPROVED** [review]

**Pull request merged**